os/include/sys/socket.h

```diff
@@ -303,8 +303,10 @@ int getsockname(int sockfd, FAR struct sockaddr *addr, FAR socklen_t *addrlen);
 */
 int getpeername(int s, struct sockaddr *name, socklen_t *namelen);
 
+#ifdef CONFIG_NET_SOCKET
```

Comment:
In `os/net/recvmsg.c`, the `#ifdef` condition is based on `CONFIG_NET`. If you believe `CONFIG_NET_SOCKET` is the correct check, then recvmsg.c also needs modification.

Reply:
1. CONFIG_NET_SOCKET can be enabled when CONFIG_NET is enabled. That means even if CONFIG_NET is enabled, CONFIG_NET_SOCKET can be disabled.
2. In os/net/socket/Make.defs which determines compilation of source code, recvmsg.c file is built when CONFIG_NET_SOCKET is enabled.
```
ifeq ($(CONFIG_NET_SOCKET),y)
SOCK_CSRCS += net_sockets.c net_close.c net_dupsd.c net_dupsd2.c
SOCK_CSRCS += net_clone.c net_vfcntl.c bsd_socket_api.c
SOCK_CSRCS += recvmsg.c sendmsg.c
endif
```
In my opinion, CONFIG_NET_SOCKET is correct and ```#ifdef CONFIG_NET``` inside recvmsg.c is redundant because of Make.defs.